sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -1162,6 +1163,166 @@ case class GpuArraysOverlap(left: Expression, right: Expression)
   }
 }
 
+case class GpuMapFromArrays(left: Expression, right: Expression) extends GpuBinaryExpression {
+
+  private val mapKeyDedupPolicy = SQLConf.get.getConf(SQLConf.MAP_KEY_DEDUP_POLICY)
+
+  override def dataType: MapType = {
+    MapType(
+      keyType = left.dataType.asInstanceOf[ArrayType].elementType,
+      valueType = right.dataType.asInstanceOf[ArrayType].elementType,
+      valueContainsNull = right.dataType.asInstanceOf[ArrayType].containsNull)
+  }
+
+  /**
+   * Compare top level offsets to ensure there are equal number of elements in
+   * keys array and values array for each row.
+   */
+  private def compareOffsets(lhs: ColumnVector, rhs: ColumnVector) : Boolean = {
+    val boolScalar = withResource(lhs.getListOffsetsView) { lhsOffsets =>
+      withResource(rhs.getListOffsetsView) { rhsOffsets =>
+        withResource(lhsOffsets.equalToNullAware(rhsOffsets)) { compareOffsets =>
+          compareOffsets.all
+        }
+      }
+    }
+    withResource(boolScalar) { all =>
+      all.getBoolean
+    }
+  }
+
+  /**
+   * Build new keys column and values column where a row is not NULL only if
+   * keys[row_id] and values[row_id] are not NULL.
+   */
+  private def nullSanitize(lhs: ColumnVector, rhs: ColumnVector) :
+  (ColumnVector, ColumnVector) = {
+    if (lhs.hasNulls || rhs.hasNulls) {
+      val combinedNulls = withResource(lhs.isNull) { lhsNulls =>
+        withResource(rhs.isNull) { rhsNulls =>
+          lhsNulls.or(rhsNulls)
+        }
+      }
+      withResource(combinedNulls) { combinedNulls =>
+            withResource(GpuScalar.from(null, left.dataType)) { leftScalar =>
+              withResource(GpuScalar.from(null, right.dataType)) { rightScalar =>
+                //  lhs: [[1,2], NULL, [3]]
+                //  rhs: [NULL, [2,5], [6]]
+                //  newLhs: [NULL, NULL, [3]]
+                //  newRhs: [NULL, NULL, [6]]
+                val newLhs = combinedNulls.ifElse(leftScalar, lhs)
+                val newRhs = combinedNulls.ifElse(rightScalar, rhs)
+                (newLhs, newRhs)
+              }
+          }
+        }
+    } else {
+      lhs.incRefCount()
+      rhs.incRefCount()
+      (lhs,rhs)
```

Comment:
optional formatting nit:
```suggestion
      (lhs, rhs)
```